### PR TITLE
Kazakhstan update

### PR DESCRIPTION
**Pull request by KJhellico**
- observed holidays
- holidays relevance period

### CODE REVIEW

File: holidays/countries/kazakhstan.py [hunk 1]
```diff
@@ -31,48 +33,64 @@ def _populate(self, year):
...
 
+        if self.observed:
+            for k, v in list(self.items()):
+                if k.weekday() in WEEKEND and k.year == year:
+                    next_workday = k + rd(days=+1)
+                    while next_workday.weekday() in WEEKEND or self.get(
+                        next_workday
+                    ):
+                        next_workday += rd(days=+1)
+                    self[next_workday] = v + " (Observed)"
```
Comment (KJhellico) [inline holidays/countries/kazakhstan.py:87]: Thess lines are not fully covered by test, although there is check with `self.holidays.observed = False`. @arkid15r, can you look into it? I tried new test functions :)

Comment (arkid15r) [reply holidays/countries/kazakhstan.py:87]: Try
```
self.assertHoliday(Kazakhstan(observed=True), *observed_holidays)
self.assertNoHoliday(Kazakhstan(observed=False), *observed_holidays)
```

Comment (KJhellico) [reply holidays/countries/kazakhstan.py:87]: It works. Thank you!

**Comment by dr-prodigy** [comment]
I unfortunately had to revert this commit, due to a failing test on an observed holiday (2012-01-03).
Could you please review and provide a fix for that @KJhellico ?
Thx as usual! 👍

**Comment by arkid15r** [comment]
The issue was caused by another [PR](https://github.com/dr-prodigy/python-holidays/blob/4008a7e3497d53fcb0dfa88a53998c1e3ff9dea9/test/common.py#L33). It was fixed in https://github.com/dr-prodigy/python-holidays/pull/844

Sorry for the inconvenience.

**Comment by dr-prodigy** [comment]
Clear @arkid15r , thank you for explaining (honest, I could not dig so much on the issue yesterday)!
I've just merged #844 , will try to re-apply this one and see if all is right now!
Thx :+1:

**Pull request merged**